src/drake/shell.clj

```diff
@@ -103,7 +103,7 @@
         proc (.exec (Runtime/getRuntime)
                     cmd-for-exec
                     ^"[Ljava.lang.String;" env
-                    (fs/file fs/*cwd*))]
+                    (fs/file (or exec-dir fs/*cwd*)))]
```

Comment:
I don't think this is possible. If `exec-dir` is nil, then `(or exec-dir fs/*cwd*)` is indistinguishable from just `fs/*cwd*`. I think the problem is not to do with the `(or)`, but rather that you've performed some side effect elsewhere that causes this. Like, probably you issued a `(cd "inner")` or whatever at some point, and then never put it back, so that the command is running in a different context. What is the command being executed that's printing `/home/chen/code/drake/sandbox/inner`? That's the only output that's surprising, and you haven't said what it's supposed to mean.